ec2utils/ec2utilsbase/lib/ec2utils/ec2preparer.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2018 SUSE LLC, Christian Bruckmayer <cbruckmayer@suse.com>
+#
+# This file is part of ec2utilsbase.
+#
+# ec2utilsbase is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ec2utilsbase is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ec2utilsbase.  If not, see <http://www.gnu.org/licenses/>.
+
+import boto3
+import os
+import random
+import datetime
+
+from ec2utils.ec2UtilsExceptions import EC2ConnectionException
+from ec2utils.ec2utils import EC2Utils
+from tempfile import mkstemp
+
+class EC2Preparer(EC2Utils):
+    """Class to prepare an Amazon EC2 account with all necessary resources"""
+
+    def __init__(self, access_key, account, key_pair_name, region, secret_key, security_group_ids, session_token, ssh_private_key_file, vpc_subnet_id, verbose):
```

Comment:
Simplify the constructor, see previous comment

Derive from EC2UtilsBase then you can take advantage of creating the connection to EC2 that is already implemented there.

Reply:
That's what I already do, no? I inherit from EC2Utils and use the _connection method from it? Or what do you mean exactly?